src/runtime/assemblymanager.cs

```diff
@@ -470,9 +471,28 @@ public static Type LookupType(string qname)
         internal static Type[] GetTypes(Assembly a)
         {
             if (a.IsDynamic)
-                return a.GetTypes();
+            {
+                try
+                {
+                    return a.GetTypes();
+                }
+                catch (ReflectionTypeLoadException exc)
+                {
+                    // Return all types that were successfully loaded
+                    return exc.Types.Where(x => x != null).ToArray();
+                }
+            }
```

Comment:
@denfromufa Is this what you had in mind?

Reply:
@filmor this is even more than what i asked for :)